### PR TITLE
Add test for import of the installed package

### PR DESCRIPTION
**Pull request by michaljurecko**
Jira: https://keboola.atlassian.net/browse/COM-482
Changes:
- Added test for user packages - installed package is imported, it was not tested before.
- Fixed module not found error for user packages, ... packages directory must already exist when starting the script.

### CODE REVIEW

File: src/transformation.py [hunk 1]
```diff
@@ -95,8 +94,6 @@ def install_packages(packages, user_option=False):
                 '--force-reinstall',
                 package
             ]
-            if user_option:
-                args.insert(6, '--user')
```
Comment (michaljurecko) [inline src/transformation.py:99]: It is already present above: https://github.com/keboola/python-transformation-v2/blob/344a441edc55defdb41816682052089a3f236798/src/transformation.py#L92

File: tests/package/config.json [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    "parameters": {
+        "packages": ["art"],
+        "blocks": [
+            {
+                "name": "block 1",
+                "codes": [
+                    {
+                        "name":"code 1",
+                        "script": [
+                            "from art import *",
```
Comment (michaljurecko) [inline tests/package/config.json:11]: Import installed package `art`.

File: Dockerfile [hunk 3]
```diff
@@ -1,6 +1,15 @@
 FROM quay.io/keboola/docker-custom-python:2.0.4
 
-RUN mkdir -p /var/www && chown www-data:www-data /var/www
+# Create director for user packages
+# This directory is usually created automatically by pip
+# ... but if it doesn't exist when you run the script
+# ... then the modules installed during the transformation are not loaded automatically!
+# ... because the loader thinks that this directory does not exist (it did not exist at the start of the script)
+RUN mkdir -p /var/www/.local/lib/python3.8/site-packages
```
Comment (pivnicek) [inline Dockerfile]: i think we should probably add here a note to change this if/when we change the python version during next update 
image
https://keboola.atlassian.net/wiki/spaces/SUP/pages/72024277/Python+R+Julia+Updates

File: tests/transformationTest.py [hunk 4]
```diff
@@ -43,6 +42,16 @@ def test_transformData(self):
...
                                  (int(row['funkyNumber']) ** 3))
+
+    def test_package(self):
+        data_dir = self.data_dir + '/package/'
+        result_file = os.path.abspath(data_dir + '/out/files/out.txt')
+        app = Transformation(data_dir)
+        app.execute()
+        with open(result_file, 'rt') as file:
+            # Cup of coffee, result of the "art" package
+            self.assertEqual('c[_]', file.read().strip())
```
Comment (michaljurecko) [inline tests/transformationTest.py:53]: Test content of the file generated by the installed package `art`.

File: docker-compose.yml [hunk 5]
```diff
@@ -6,9 +6,9 @@ services:
     user: www-data
     tty: true
     stdin_open: true
-    command: bash
```
Comment (michaljurecko) [inline docker-compose.yml:9]: .. so we can try default command in local dev env

**Review by pivnicek — APPROVED** [review]
Great!  I love the test :-)

File: Dockerfile (hunk 3 above)
Comment (michaljurecko) [reply Dockerfile]: very good note, thank you! ...  I'd rather get that path dynamically by `python -c "import site; print(site.USER_SITE)"` (as `www-data` user)

Comment (pivnicek) [reply Dockerfile]: yes, that would be ideal

**Comment by michaljurecko** [comment]
@marc please review the new commit: https://github.com/keboola/python-transformation-v2/pull/7/commits/f46e28749185ebc10ff68e1aca21442c18ccf41f

File: Dockerfile [hunk 6]
```diff
@@ -1,6 +1,16 @@
 FROM quay.io/keboola/docker-custom-python:2.0.4
 
-RUN mkdir -p /var/www && chown www-data:www-data /var/www
+# Create directory for user packages
+# This directory is usually created automatically by pip
+# ... but if it doesn't exist when you run the script
+# ... then the modules installed during the transformation are not loaded automatically!
+# ... because the loader thinks that this directory does not exist (it did not exist at the start of the script)
+# Eg. mkdir -p /var/www/.local/lib/python3.8/site-packages
+RUN mkdir -p $(su www-data -s /bin/bash -c "python -c 'import site; print(site.USER_SITE)'")
```
Comment (pivnicek) [inline Dockerfile:9]: 👍

**Review by pivnicek — APPROVED** [review]
parada

**Pull request merged**